modules/LookUpCatch.cmake

```diff
@@ -0,0 +1,66 @@
+# Installs catchorg/Catch2 into build directory
+#
+# - URL: defaults to latest single_include
+# - VERSION: defaults to latest
+
+if(Catch_ARGUMENTS)
+  cmake_parse_arguments(Catch "" "URL;VERSION" ""
+    ${Catch_ARGUMENTS})
+endif()
+
+if(NOT Catch_VERSION)
+  set(Catch_URL_VERSION master)
+else()
+  set (Catch_URL_VERSION "v${Catch_VERSION}")
+endif()
+
+if(NOT Catch_URL)
+  set(Catch_URL https://raw.githubusercontent.com/catchorg/Catch2/${Catch_URL_VERSION}/single_include/catch2/catch.hpp)
+endif()
+
+set(Catch_FILE "${EXTERNAL_ROOT}/include/catch.hpp")
+file(MAKE_DIRECTORY "${EXTERNAL_ROOT}/include")
+file(DOWNLOAD ${Catch_URL} "${Catch_FILE}")
+
+file(READ "${Catch_FILE}" CATCHSTRING LIMIT 1000)
```

Comment:
You should add a comment explaining what this next section is doing (checking download worked by ensuring the file is long enough).

Reply:
Yeah... that was on the original file. I imagine is checking whether the file is long enough as I suppose redirects or errors will be shorter ones.